src/main/java/com/employeesService/EmployeesService/service/SalaryServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.employeesService.EmployeesService.service;
+
+import com.employeesService.EmployeesService.exception.EmployeeNotFoundException;
+import com.employeesService.EmployeesService.model.Employee;
+import com.employeesService.EmployeesService.model.Salary;
+import com.employeesService.EmployeesService.repository.EmployeeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class SalaryServiceImpl implements SalaryService {
+
+    private final EmployeeRepository employeeRepository;
+
+    @Autowired
+    public SalaryServiceImpl(EmployeeRepository employeeRepository) {
+        this.employeeRepository = employeeRepository;
+    }
+
+
+    @Override
+    public Salary assignSalaryToEmployee(Long empNo, Salary salaryToAdd) {
+        Employee employee = this.employeeRepository.findById(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
+        salaryToAdd.setEmpNo(empNo);
+        salaryToAdd.setSalary(salaryToAdd.getSalary());
+        salaryToAdd.setFromDate(salaryToAdd.getFromDate());
+        for (int i = 0; i < employee.getSalaries().size(); i++) {
+            if (employee.getSalaries().get(i).getToDate() == null) {
+                employee.getSalaries().get(i).setToDate(salaryToAdd.getFromDate());
+            }
+        }
+        salaryToAdd.setToDate(salaryToAdd.getToDate());
+        employee.assignSalary(salaryToAdd);
+        this.employeeRepository.save(employee);
+        return salaryToAdd;
+    }
+
+    @Override
+    public List<Salary> listAllSalaries(Long empNo) {
+        Employee employee = this.employeeRepository.findByIdWithSalary(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
+        return employee.getSalaries();
```

Comment:
Does this fail? Normally it should 🤔 

Did you disable: `spring.jpa.open-in-view`

Set `spring.jpa.open-in-view=false`